ksa_compliance/standard_doctypes/sales_invoice.py

```diff
@@ -101,3 +101,11 @@ def should_enable_zatca_for_invoice(invoice_id: str) -> bool:
 
     posting_date = frappe.db.get_value('Sales Invoice', invoice_id, 'posting_date')
     return posting_date >= start_date
+
+
+def calculate_tax_amount(self):
+    if self.items:
+        for item in self.items:
+            item_tax_rate = frappe.get_value("Item Tax Template Detail", {"parent": item.item_tax_template}, ['tax_rate'])
```

Comment:
This logic doesn't take multiple rates/charges per item into account (The `Item Tax Template` may have multiple `Item Tax Template Detail` lines). This will work fine for simple cases with one line, but will work incorrectly for others; it'll pick the tax_rate of the first line most likely.

We need to discuss this point with the business team:
* Whether it's OK to do this for now, or if we should at least fail with an error for unexpected cases (i.e. more than one line)
* Whether we should implement the full solution, and what it entails (I assume getting all lines whose account is type "Tax" and summing the rates, but I could be wrong)